### PR TITLE
python 3 support

### PR DESCRIPTION
**Pull request by olatarkowska**
* Migrate to python 3
 * Change ownership of venv
 * Change location
 * reduce number of dependencies

### CODE REVIEW

File: ansible/templates/centos7/deps_ice3.6.sh.j2 [hunk 1]
```diff
@@ -9,4 +9,7 @@ set -e -u -x
 {% endif %}
 
 # option 1 and option 2
-{{ virtualenv_path }}/bin/pip install --upgrade https://github.com/ome/zeroc-ice-py-centos7/releases/download/0.1.0/zeroc_ice-3.6.4-cp27-cp27mu-linux_x86_64.whl
+. {{ virtualenv_path }}/bin/activate
+pip install --upgrade pip
+
+pip install --upgrade https://github.com/ome/zeroc-ice-py-manylinux/releases/download/0.1.0/zeroc_ice-3.6.5-cp36-cp36m-manylinux2010_x86_64.whl
```
Comment (manics) [inline ansible/templates/centos7/deps_ice3.6.sh.j2]: As discussed in slack this includes a static openssl library so shouldn't be used in production. If this is for testing only and not for the production docs this is fine.

File: ansible/templates/common/install_whitenoise.sh.j2 [hunk 2]
```diff
@@ -5,6 +5,8 @@ Install `WhiteNoise <http://whitenoise.evans.io/>`_::
 
 set -e -u -x
 
+export OMERODIR=$OMERODIR
```
Comment (manics) [inline ansible/templates/common/install_whitenoise.sh.j2]: Is this needed? `omero` isn't run in this script.

File: ansible/templates/debian9/deps_ice3.6.sh.j2 [hunk 3]
```diff
@@ -9,4 +9,5 @@ set -e -u -x
 {% endif %}
 
 # option 1 and option 2
-{{ virtualenv_path }}/bin/pip install --upgrade https://github.com/ome/zeroc-ice-py-debian9/releases/download/0.1.0/zeroc_ice-3.6.4-cp27-cp27mu-linux_x86_64.whl
+{{ virtualenv_path }}/bin/pip3 install --upgrade pip
+{{ virtualenv_path }}/bin/pip3 install --upgrade https://github.com/ome/zeroc-ice-py-manylinux/releases/download/0.1.0/zeroc_ice-3.6.5-cp36-cp36m-manylinux2010_x86_64.whl
```
Comment (manics) [inline ansible/templates/debian9/deps_ice3.6.sh.j2]: Same as earlier comment concerning the static `openssl` library

File: ansible/templates/centos7/deps_ice3.6.sh.j2 (hunk 1 above)
Comment (jburel) [inline ansible/templates/centos7/deps_ice3.6.sh.j2]: to match what was previously there 
https://github.com/ome/zeroc-ice-py-centos7/releases/download/0.2.1/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl should work for py3

**Comment by olatarkowska** [comment]
Fixed now

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

**Comment by jburel** [comment]
Discussed yesterday: we will keep the "default" python 3 version (3.5) for ubuntu16.04 and debian 9  and indicate that we provide instructions but not run daily tests with those OS

**Comment by will-moore** [comment]
Going through https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-docs/lastSuccessfulBuild/artifact/omero/_build/html/sysadmins/unix/install-web/walkthrough/omeroweb-install-centos7-ice3.6.html the omero user needs to do
```source /opt/omero/web/venv/bin/activate``` before you can run ```omero...``` e.g. ```omero config set omero.web.application_server wsgi-tcp```

**Comment by will-moore** [comment]
```(venv) [omero@f39a61c6ff23 /]$ omero web config nginx --http "${WEBPORT}" --servername "${WEBSERVER_NAME}" > /home/omero/nginx.conf.tmp
Error printing text
(venv) [omero@f39a61c6ff23 /]$ omero web config nginx --http 80 --servername localhost > /home/omero/nginx.conf.tmp
Error printing text```

**Comment by will-moore** [comment]
```
$ omero config set omero.web.application_server wsgi-tcp
/opt/omero/web/venv/lib64/etc/grid not found; using /home/omero/omero/config.xml
```
With https://github.com/ome/omero-py/pull/14 it will be essential to set ```$OMERODIR``` before config etc.

**Comment by will-moore** [comment]
Probably because of the errors above, I eventually see:
```
[root@f39a61c6ff23 /]# systemctl start nginx
Job for nginx.service failed because the control process exited with error code. See "systemctl status nginx.service" and "journalctl -xe" for details.
```

**Comment by will-moore** [comment]
Probably just didn't read this carefully yesterday, but noticed this is a bit confusing "...a few more dependencies will be required (option 2):" since no option 1 or 2 has been mentioned before then. Maybe "(see option 2 below):". And then in the code (instead of ```# To install OMERO.web using option 2``` which sounds like this is doing the omero-web install, maybe:
```# Extra dependencies if using option 2 below```

**Comment by jburel** [comment]
@will-moore thanks for pointing that out. Actually I have simplified it since we no longer need the option and unifies the doc with omero.server

File: ansible/templates/common/virtualenv.sh.j2 [hunk 4]
```diff
@@ -1,17 +1,13 @@
 {% if doc %}
-Create the virtual environment. This is the preferred way to install OMERO.web::
+Create the virtual. This is the recommended way to install OMERO.web::
```
Comment (will-moore) [inline ansible/templates/common/virtualenv.sh.j2]: Create the virtual ?

File: ansible/templates/common/introduction.j2 [hunk 5]
```diff
@@ -1,4 +1,3 @@
 {% if doc %}
-This is an example walkthrough for installing OMERO.web decoupled from the OMERO.server in a **virtual environment** using a dedicated system user. Installing OMERO.web in a virtual environment is the preferred way. For convenience in this walkthrough, we will use the **{{ omero_user }} system user** and define the main OMERO.web configuration options as environment variables.
-We recommend that you install OMERO.web in a different virtual environment that the one used to install ``omero-py``.
+This is an example walkthrough for installing OMERO.web decoupled from the OMERO.server in a **virtual environment** using a dedicated system user. Installing OMERO.web in a virtual environment is the preferred way. For convenience in this walkthrough, we will use the **{{ omero_user }} system user** and define the main OMERO.web configuration options as environment variables. We recommend that you install OMERO.web in a different virtual environment that the one used to install ``omero-py``.
```
Comment (will-moore) [inline ansible/templates/common/introduction.j2]: different virtual environment *that* -> than

Comment (will-moore) [reply ansible/templates/common/introduction.j2]: But that OMERO.web vv omero-py sentence is confusing. Surely omero-web includes omero-py?

**Comment by joshmoore** [comment]
Merging along with https://github.com/ome/ome-documentation/pull/2038#issuecomment-561349685

File: ansible/group_vars/centos7-ice3.6 [hunk 6]
```diff
@@ -30,15 +30,13 @@ default:
 
 omero_user: "omero"
 omero_user_home_dir: "/home/{{ omero_user }}"
-virtualenv_path: "{{ omero_user_home_dir }}/omerowebvenv"
+virtualenv_path: "/opt/omero/web/venv"
```
Comment (joshmoore) [inline ansible/group_vars/centos7-ice3.6]: @jburel : thoughts on moving this to `venv3` to match the ansible role?

Comment (jburel) [reply ansible/group_vars/centos7-ice3.6]: I will open follow up PR for omero-install and omeroweb-install

Comment (jburel) [reply ansible/group_vars/centos7-ice3.6]: PRs opened:
* https://github.com/ome/omero-install/pull/217
* https://github.com/ome/omeroweb-install/pull/50

**Pull request merged**